### DISCUSSION TITLE
Hyphens in field names should give warning to people using GraphQL

### DISCUSSION THREAD

**Discussion by TomDo1234** [payloadcms/payload#1826] [Roadmap]
# Bug Report

GraphQL doesn't work with field names that have hyphens in them, This lead to me doing a database conversion and renaming all my fields to camelcase. This should be a warning in the TSConfig or something. Or ESLint or whatever is able to give a warning when using a field name with a hyphen or anything that would break GraphQL

## Steps to Reproduce

1. Create a collection and add a field with a name that has a hyphen in it, or kebab case
2. GraphQL doesn't work with it

**Comment by DanRibbens** [comment]
I don't think a warning is necessary if we instead format field names to replace invalid characters.

That is already done on slugs. I'd like to see what it breaks to do try it.

**Comment by TomDo1234** [comment]
Seems awesome Dan, I thought it was too hard to format.

But if we CAN format it, that would be great.

**Comment by jmikrut** [comment]
Gonna convert this to a discussion and add it to our roadmap. Thanks for the discussions here!

👍

**Comment by JarrodMFlesch** [comment]
Was previously fixed in [this PR](https://github.com/payloadcms/payload/pull/11556).
